feat: first transition added for the image preview instead of loading…

MasterMindNET opened the pull request:
… animtion; related gallery PR:64

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [ ] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [x] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Please also see the gallery reltated PR: [PR link](https://github.com/humhub-contrib/gallery/pull/64)

luke- commented on static/js/humhub/humhub.ui.gallery.js:
```diff
@@ -6,24 +6,59 @@
                 options['urlProperty'] = 'src';
             }
             blueimp.Gallery($links.get(), options);
+
+            var $slides = $('#blueimp-gallery .slides');
+
+            var animatePreviewForFirstTime = function () {
+                $slides.css({'opacity': 0.1});
+                var $firstImage = $slides.find('.slide img') ? $slides.find('.slide img')[0] : '';
+
+                if ($firstImage !== '' && $firstImage.complete) {
+                    var firstAnimationTimeInMs = 1250;
+
+                    $slides.fadeTo(firstAnimationTimeInMs, 1);
+                    firstImageIsLoaded = true;
+                } else {
+                    $slides.css({'opacity': 1});
+                }
+            }
+            if (!firstImageIsLoaded) {
+                animatePreviewForFirstTime();
+            }
+
```
Please remove dead code

MasterMindNET replied:
I have just done it.

buddh4 commented:
As I understand you've removed image loading indicator, and added a fade animation once the image is loaded. But until the image is loaded we only see a black background. Not sure if the user experienced is improved by this. Also its a rather big block of code for a small effect ;)

buddh4 commented:
Can you elaborate a bit more what problems do you see with the old behavior, which are soved by this PR.

MasterMindNET commented:
> As I understand you've removed the image loading indicator, and added a fade animation once the image is loaded. But until the image is loaded we only see a black background. Not sure if the user experience is improved by this. Also its a rather big block of code for a rather small effect ;)

Good day, @buddh4!

Previous behavior was some kind of improper loading of image, it was loading line by line; But now, I added a slow fades in during 1.25 seconds.
I have asked in [this PR](https://github.com/humhub-contrib/gallery/pull/64#issuecomment-683281616)
How I can find the wrong part? Because even the old loader was not spinning, so there is something wrong with it.

Could you please help me to find why it is not spinning, and how do you see it should be? 
I should only fix the spin issue of this loader or something else too?
I try to fix the loader one more time, but I tried to do it on the jsfiddle (on just a clean blueimp gallery), and it was working well.

Any ideas would help, thank you!

luke- commented:
@buddh4 Should we should merge this PR oder close it for now?

buddh4 commented:
I'll close it. The issue seems to be related with your environment. The image of blueimp should be a gif, not sure why its not running in your tests.
